tempodb/encoding/vparquet2/block_traceql.go

```diff
@@ -1757,19 +1753,19 @@ type traceCollector struct {
 
 var _ parquetquery.GroupPredicate = (*traceCollector)(nil)
 
+func newTraceCollector() *traceCollector {
```

Comment:
the comments in this PR have been really helpful for me to understand some of the more esoteric lines. maybe add a comment here about the role of the tracecollector in creating spansets?